### ISSUE TITLE
Add a new action for mass symlinking

### ISSUE THREAD

**Issue by Szelethus** [whisperity/dotfiles-framework#6]
It is a common pattern in packages that we symlink every file found in a `bin/` folder to `~/bin/`, as seen in https://github.com/whisperity/dotfiles/blob/master/packages/cmake/package.yaml:

```yaml
install:
  # ...
  - action: shell
    command: "cd $HOME/bin; ln -s ../opt/cmake/bin/* ."
uninstall:
  - action: shell
    command: "cd $HOME/bin; for sl in $(ls ../opt/cmake/bin); do rm -f ${sl}; done"
```

It would be great to dedicate a new command to it.

**Comment by whisperity** [comment]
Yeah, I've run into wanting to commit back changes to my files to the main package descriptor repository without needing to do the reverse copy of the files. Locally I'm now running multiple machines where I hotwired the entire Dotfiles system and instead created the symlinks manually, but setting them up is a massive tedium, and of course, why would we have a framework if there is no point in using that (and lo and behold, the main developer of it **isn't using it daily**? WTF?).

I've been cooking up some ideas, and I think I got the design accurate for now. First, there will be a `symlink` action kind just like `copy`, exact same interface and whatnot, but will actually explicitly create symlinks no matter what. (The inverse operation of the symlink is trivially a delete.)

Moreover, I think I'll introduce a flag, something like `--always-symlink` into the install process, in which case **every `copy` operation** (okay, *almost* every, because there are a few edge cases, e.g., when copying from the `$TEMPORARY_DIR`) will in fact behave *as-if* it was a `symlink` operation, implicitly. Also, add a new entry to the schema where the package author can force-disable this option for a specific step. (The details for this are a bit foggy, still.)

This way, scripts that deal with explicit symlinks would be simplified as requested, and deployments where you would want to actively *version* changes to your configuration files (as opposed to just fire-and-forget a setup), will support this notion out-of-the-box.
